Modules/Scripted/DICOMLib/DICOMPlugin.py

```diff
@@ -140,19 +156,48 @@ def export(self,exportable):
     """
     return ""
 
-  def defaultSeriesNodeName(self,seriesUID):
+  def getVolumeName(self, files):
     """Generate a name suitable for use as a mrml node name based
-    on the series level data in the database"""
-    instanceFilePaths = slicer.dicomDatabase.filesForSeries(seriesUID)
-    if len(instanceFilePaths) == 0:
-      return "Unnamed Series"
-    seriesDescription = slicer.dicomDatabase.fileValue(instanceFilePaths[0],self.tags['seriesDescription'])
-    seriesNumber = slicer.dicomDatabase.fileValue(instanceFilePaths[0],self.tags['seriesNumber'])
-    name = seriesDescription
-    if seriesDescription == "":
-      name = "Unnamed Series"
-    if seriesNumber != "":
-      name = seriesNumber + ": " + name
+    on the series level data in the database.
+
+    Arguments
+
+      files: files that will be used to build the volume name using the DICOM tags in a list of DICOM files
+    """
+    if not files or len(files) == 0:
+      return "Unknown"
+
+    if not self.volumeNameTagsTemplate:
+      # Reset to default template
+      self.resetVolumeNameTagsTemplate()
+
+    # Get the volume name based on the current template
+    name = self.volumeNameTagsTemplate
+    # Replace evey tag found in the template, using a regular expression that will search for this pattern: @DICOM_TAG@
+    pattern = ".*?@(.+?)@.*?"
+    for expr in re.finditer(pattern, name):
+      dicomTagCode = expr.groups()[0]
+      try:
+        # Change the first letter to lowercase in order that it matches the internal key (CTK DICOM Editor shows the
+        # tags capitalizing each word)
+        fixedTag = dicomTagCode[0].lower() + dicomTagCode[1:]
+        # Remove the possible parenthesis
+        if fixedTag.startswith("(") and fixedTag.endswith(")"):
+          fixedTag = fixedTag[1:-1]
+        if self.tags.has_key(fixedTag):
+          # Instead of a code, the tag is one of the common well known DICOM tags (ex: seriesNumber)
+          fixedTag = self.tags[fixedTag]
+
+        # Search for the real value in the first image of the series
+        codeValue = slicer.dicomDatabase.fileValue(files[0], fixedTag)
```

Comment:
Note that values can contain special characters that can cause problems. There are 2 places where filtering for special characters is done: 

1. https://github.com/pieper/dicomsort/blob/master/dicomsort.py#L99-L108
2. (simpler) https://github.com/naucoin/CTK/blob/86837e6a68a0518f045be0ec93006af5500140f9/Libs/DICOM/Widgets/ctkDICOMBrowser.cpp#L957-L961

See related discussion in http://www.na-mic.org/Bug/view.php?id=3163